docs/content/posts/2024-10-01-release-2.0-preview.md

```diff
@@ -0,0 +1,1264 @@
+- `org.apache.flink.state.api.SavepointWriter`
+  - method removed:
+    - `org.apache.flink.state.api.SavepointWriter fromExistingSavepoint(java.lang.String)`
+    - `org.apache.flink.state.api.SavepointWriter fromExistingSavepoint(java.lang.String, org.apache.flink.runtime.state.StateBackend)`
+    - `org.apache.flink.state.api.SavepointWriter newSavepoint(int)`
+    - `org.apache.flink.state.api.SavepointWriter newSavepoint(org.apache.flink.runtime.state.StateBackend, int)`
+    - `org.apache.flink.state.api.SavepointWriter removeOperator(java.lang.String)`
+    - `org.apache.flink.state.api.SavepointWriter withOperator(java.lang.String, org.apache.flink.state.api.StateBootstrapTransformation<T>)`
+- `org.apache.flink.state.api.SavepointWriterOperatorFactory`
+  - method modified:
+    - `org.apache.flink.streaming.api.operators.StreamOperatorFactory<org.apache.flink.state.api.output.TaggedOperatorSubtaskState><org.apache.flink.state.api.output.TaggedOperatorSubtaskState> (<-org.apache.flink.streaming.api.operators.StreamOperator<org.apache.flink.state.api.output.TaggedOperatorSubtaskState><org.apache.flink.state.api.output.TaggedOperatorSubtaskState>) createOperator(long, org.apache.flink.core.fs.Path)`
+
+
+## List of removed configuration options
+
+- cluster.evenly-spread-out-slots
+- fine-grained.shuffle-mode.all-blocking
+- high-availability.job.delay
+- high-availability.zookeeper.path.running-registry
+- jobmanager.adaptive-batch-scheduler.avg-data-volume-per-task
+- jobmanager.adaptive-batch-scheduler.default-source-parallelism
+- jobmanager.adaptive-batch-scheduler.max-parallelism
+- jobmanager.adaptive-batch-scheduler.min-parallelism
+- jobmanager.adaptive-batch-scheduler.speculative.block-slow-node-duration
+- jobmanager.adaptive-batch-scheduler.speculative.enabled
+- jobmanager.adaptive-batch-scheduler.speculative.max-concurrent-executions
+- jobmanager.heap.mb
+- jobmanager.heap.size
+- local.number-resourcemanager
+- pipeline.auto-type-registration
+- pipeline.default-kryo-serializers
+- pipeline.registered-kryo-types
+- pipeline.registered-pojo-types
+- resourcemanager.taskmanager-release.wait.result.consumed
+- security.kerberos.fetch.delegation-token
+- security.kerberos.tokens.renewal.retry.backoff
+- security.kerberos.tokens.renewal.time-ratio
+- security.ssl.enabled
+- slotmanager.taskmanager-timeout
+- sql-client.display.max-column-width
+- state.backend.async
+- state.backend.latency-track.history-size
+- state.backend.latency-track.keyed-state-enabled
+- state.backend.latency-track.sample-interval
+- state.backend.latency-track.state-name-as-variable
+- state.backend.local-recovery
+- state.backend.type
+- table.exec.legacy-transformation-uids
+- table.exec.shuffle-mode
+- table.optimizer.source.aggregate-pushdown-enabled
+- table.optimizer.source.predicate-pushdown-enabled
```

Comment:
Additionally, several table-related configurations have also been removed (the corresponding ConfigOption#withDeprecatedKeys method has been deleted on those configs):
- table.exec.topn-cache-size
- table.exec.deduplicate.insert-and-updateafter-sensitive.enabled
- table.exec.deduplicate.mini-batch.compact-changes.enabled
- table.optimizer.sql-to-rel.project.merge.enabled

Reply:
Removed key in `withDeprecatedKeys` methods are added.